src/ir/ranges.cpp

```diff
@@ -352,6 +352,12 @@ namespace phylanx { namespace ir
                 std::int64_t const step = int_range.step();
 
                 std::int64_t const distance_to_start = stop - start;
+
+                // Adding step to start must get closer to stop
+                if ((distance_to_start > 0) != (step > 0))
```

Comment:
Shouldn't this include comparing against `0`? e.g. something like:
```
if ((distance_to_start >= 0) != (step >= 0))
```

Reply:
It doesn't need to compare against `0`. I added a test case that explicitly tests that case. `step == 0` was not handled anywhere else. It is handled now.